src/ClientBuilder.php

```diff
@@ -68,127 +68,97 @@ final class ClientBuilder implements ClientBuilderInterface
     public function __construct(Options $options = null)
     {
         $this->options = $options ?? new Options();
+
+        $this->httpClient = $this->options->getHttpClient() ?? new HttpClient($this->sdkIdentifier, $this->sdkVersion);
+        $this->transport = $this->options->getTransport() ?? new HttpTransport(
+            $this->options,
+            $this->httpClient,
+            new PayloadSerializer($this->options),
+            $this->logger
+        );
```

Comment:
We should move this down to the `getClient()` which is the actual builder method. And only create a client or use the one from the options if the transport was not set. Also saves needing the `httpClient` instance variable.

Reply:
Moved to https://github.com/getsentry/sentry-php/issues/1595